test/unit/test_bigquery_adapter.py

```diff
@@ -652,17 +652,123 @@ def test_parse_partition_by(self):
                 "field": "ts",
             }).to_dict(), {
                 "field": "ts",
-                "data_type": "date"
+                "data_type": "date",
+                "granularity": "DAY"
```

Comment:
This one's on me for asking you to lowercase:
```suggestion
                "granularity": "day"
```